experiments/pipeline/examples/general_usage.md

```diff
@@ -0,0 +1,195 @@
+2. only when successful more data is extracted
+3. only when successful newly extracted data is normalized and loaded.
+
+extract / normalize / load are atomic. the `run` is as close to be atomic as possible.
+
+the `run` and `load` return information on loaded packages: to which datasets, list of jobs etc. let me think what should be the content
+
+> `load` is atomic if SQL transformations ie in `dbt` and all the SQL queries take into account only committed `load_ids`. It is certainly possible - we did in for RASA but requires some work... Maybe we implement a fully atomic staging at some point in the loader.
+
+
+## the `Pipeline` object
+There are many ways to create or get current pipeline object.
+```python
+
+# create and get default pipeline
+p1 = dlt.pipeline()
+# create explicitly configured pipeline
+p2 = dlt.pipeline(name="pipe", destination=bigquery)
+# get recently created pipeline
+assert dlt.pipeline() is p2
+# load data with recently created pipeline
+assert dlt.run(source=taktile_data()) is p2
+assert taktile_data().run() is p2
+
+```
+
+The `Pipeline` object provides following functionalities:
+1. `run`, `extract`, `normalize` and `load` methods
+2. a `pipeline.schema` dictionary-like object to enumerate and get the schemas in pipeline
+3. schema get with `pipeline.schemas[name]` is a live object: any modification to it is automatically applied to the pipeline with the next `run`, `load` etc. see [working_with_schemas.md](working_with_schemas.md)
+4. it returns `sql_client` and `native_client` to get direct access to the destination (if destination supports SQL - currently all of them do)
+5. it has several methods to inspect the pipeline state and I think those should be exposed via `dlt pipeline` CLI
+
+for example:
+- list the extracted files if any
+- list the load packages ready to load
+- list the failed jobs in package
+- show info on destination: what are the datasets, the current load_id, the current schema etc.
+
+
+## Examples
+
+Loads data from `taktile_data` source function into bigquery. All the credentials and configs are taken from the config and secret providers.
+
+Script was run with `python taktile.py`
+
+```python
+from my_taktile_source import taktile_data
+from dlt.destinations import bigquery
+
+# the `run` command below will create default pipeline and use it to load data
+# I only want logs from the resources present in taktile_data
+taktile_data.select("logs").run(destination=bigquery)
+
+# alternative
+dlt.run(source=taktile_data.select("logs"))
+```
+
+Explicitly configure schema before the use
+```python
+import dlt
+from dlt.destinations import bigquery
+
+@dlt.source
+def data(api_key):
+    ...
+
+
+dlt.pipeline(name="pipe", destination=bigquery, dataset="extract_1")
+# use dlt secrets directly to get api key
+# no parameters needed to run - we configured destination and dataset already
+data(dlt.secrets["api_key"]).run()
```

Comment:
interesting but I think we wanna parametrise the changing stuff and freeze names on the function calls?

Reply:
this one I do not understand?